cloud/libvirt/actuators/machine/utils/domain.go

```diff
@@ -467,19 +457,28 @@ func CreateDomain(name, ignKey, volumeName, hostName, networkInterfaceName, netw
 
 	log.Printf("[INFO] setCoreOSIgnition")
 	if ignKey != "" {
-		if err := setCoreOSIgnition(&domainDef, ignKey); err != nil {
+		ignVolume, err := getVolumeFromPool(ignKey, poolName, virConn)
+		if err != nil {
+			return fmt.Errorf("error getting ignition volume: %v", err)
+		}
+		ignVolumePath, err := ignVolume.GetPath()
+		if err != nil {
+			return fmt.Errorf("error getting ignition volume path: %v", err)
+		}
+		if err := setCoreOSIgnition(&domainDef, ignVolumePath); err != nil {
 			return err
 		}
 	} else {
 		return fmt.Errorf("machine does not has a IgnKey value")
 	}
 
 	log.Printf("[INFO] setDisks")
-	VolumeKey := baseVolumePath + volumeName
-	if volumeName == "" {
-		volumeName = name
+	diskVolume, err := getVolumeFromPool(volumeName, poolName, virConn)
+	if err != nil {
+		return fmt.Errorf("can't retrieve volume %s for pool %s: %v", volumeName, poolName, err)
```

Comment:
If [`getVolumeFromPool` becomes consistent about listing the volume and pool in its error messages][1], we can drop them here.

[1]: https://github.com/openshift/cluster-api-provider-libvirt/pull/45/files#r226850669